lago/export.py

```diff
@@ -0,0 +1,239 @@
+        """
+        with LogTask('Making disk sparse'):
+            # Removed unused space from the disk
+            utils.sparse(self.dst, self.disk['format'])
+
+    def calc_sha(self, checksum):
+        """
+        Calculate the checksum of the new exported disk, write it to
+        a file, and update this managers 'exported_metadata'.
+
+        Args:
+            checksum(str): The type of the checksum
+        """
+        with LogTask('Calculating {}'.format(checksum)):
+            with open(self.dst + '.hash', 'wt') as f:
+                sha = utils.get_hash(self.dst, checksum)
+                f.write(sha)
+            self.exported_metadata[checksum] = sha
+
+    def compress(self):
+        """
+        Compress the new exported image,
+        Block size was taken from virt-builder page
+        """
+        if not self.do_compress:
+            return
+        with LogTask('Compressing disk'):
+            utils.compress(self.dst, 16777216)
+            os.unlink(self.dst)
+
+    def update_lago_metadata(self):
+        with LogTask('Updating Lago metadata'):
+            self.exported_metadata['size'] = os.stat(self.dst).st_size
+            self.exported_metadata['name'] = self.name
+            self.exported_metadata['version'] = time.strftime("%Y%m%d.0")
+
+    def write_lago_metadata(self):
+        with LogTask('Writing Lago metadata'):
+            with open(self.dst + '.metadata', 'wt') as f:
+                json.dump(self.exported_metadata, f)
+
+
+class TemplateExportManager(DiskExportManager):
+    """
+    TemplateExportManager is responsible exporting images of type template.
+
+    Attributes:
+        standalone (bool): If true, create a new image which is the result of
+         merging all the layers of src (the image that we want to export).
+        src_qemu_info (dict): Metadata on src which was generated by qemu-img.
+    """
+
+    def __init__(self, dst, disk_type, disk, do_compress, *args, **kwargs):
+        super(TemplateExportManager, self).__init__(
+            dst, disk_type, disk, do_compress
+        )
+        self.standalone = kwargs['standalone']
+        self.src_qemu_info = utils.get_qemu_info(self.src, backing_chain=True)
+
+    def rebase(self):
+        """
+        Change the backing-file entry of the exported disk.
+        Please refer to 'qemu-img rebase' manual for more info.
+        """
+        if len(self.src_qemu_info) == 1:
+            # Base image (doesn't have predecessors)
+            return
+
+        if self.standalone:
+            # Consolidate the layers and base image
+            utils.qemu_rebase(target=self.dst, backing_file="")
+        else:
+            if len(self.src_qemu_info) > 2:
+                LOGGER.error(
+                    'Layered export is currently supported for one '
+                    'layer only.  You can try to use Standalone export.'
+                )
+                sys.exit(1)
```

Comment:
I agree about the user-experience, but I don't think this is easily solved by ``sys.exit``:
1. The rest of the code(I think) doesn't do that, which will cause inconsistencies when trying to debug issues. 
2. The 'rebase' command is not a direct CLI command, other functions may call it in different scenarios, not even necessarily as in this PR, so there will be no way to know if they are responsible for a wrong call, or it exits as expected here. 
3. We might have right now(didn't check) or have in the future cleanup code that should be called before the program terminates. Simply exiting the program abruptly might prevent it from running (python raises ``SystemExit`` exception on ``sys.exit`` so I think 'finally' blocks up the stracktrace should be executed, but other stuff might not)
4. Following 3- I'm not sure how python will behave as far as memory cleanup and object releases when issued sys.exit(1), I would guess right now its not a big issue, but might be.
5. There is no logging of the actual exception and its stacktrace. 

If you still want to change this user behaviour, an alternative that comes to mind:

Add in the export function(the direct call the user makes) in  ``cmd.py`` something like:
```python
try:
    ... do your stuff here
catch (your expected exception list) as e:
   LOGGER.info(e.message)
   LOGGER.exception(e)
   sys.exit(1)
```
I can't say I'm 100% pleased with that, as we still swallow the exception(or more correctly raise a different one - ``SystemExit``, instead of the original one), but at least it will be logged to ```lago.log``` and done as close as possible 'up' to the stacktrace, so whoever wanted to clean-up his mess had a chance to do so. 


Further ahead: we can do the same technique for all exception, maybe add a flag that will control whether to print the stacktrace, if set to false, print the message to stdout, and log the entire exception to lago.log. Alternatively, we can control the depth of the printed stacktrace with ```traceback.print_tb```